src/Abstractions/RetryPolicy.cs

```diff
@@ -123,11 +126,56 @@ public class RetryPolicy
     /// </value>
     public TimeSpan RetryTimeout { get; }
 
+    /// <summary>
+    /// Gets a delegate to call on exception to determine if retries should proceed.
+    /// </summary>
+    /// <value>
+    /// Defaults delegate that always returns true (i.e., all exceptions are retried).
+    /// </value>
+    public Func<Exception, bool> Handle { get; private init; }
```

Comment:
Not sure I like this being public - it is a bit misleading. Making it `internal` though will require moving other parts around. Can you add `[EditorBrowsable(EditorBrowsableState.Never)]` to this?

Reply:
Left as public and added the attribute.